python/cuml/dask/decomposition/base.py

```diff
@@ -28,13 +28,13 @@
 from cuml.dask.common.input_utils import DistributedDataHandler
 
 
-class BaseDecompositionFitMixin(BaseEstimator):
+class BaseDecomposition(BaseEstimator):
```

Comment:
Question / not review blocker: Should we add Dask models to the pickle tests?

Reply:
I certainly think we have done enough consolidation now, and written enough of these models, to be able to think about pickling them. I have been kicking around a fairly simple design in my head that can even further fuse the single GPU and distributed APIs together, provide a FIL-like inference capability for just about all of our single GPU models, and still make it very simple for users. 

I created https://github.com/rapidsai/cuml/issues/1841 for feedback and further discussion. My thinking is that this new functionality might be better tested inside `/tests/dask/test_pickle.py`. I would love to have your feedback on the proposed design!